### DISCUSSION TITLE
Websocket proxy help

### DISCUSSION THREAD

**Discussion by lostb1t** [salvo-rs/salvo#336] [Q&A]
struggling to get the web socket proxy working. Code is copy paste from the example. ws endpoint is a plex server. But tried with other endpoints

```
use salvo::prelude::*;
use salvo::proxy::Proxy;

#[tokio::main]
async fn main() {
    tracing_subscriber::fmt().init();

    let router = Router::new()
        .push(Router::with_path("<**ws>").handle(Proxy::new("http://46.4.30.217:42405")));

    let acceptor = TcpListener::new("0.0.0.0:5800").bind().await;
    Server::new(acceptor).serve(router).await;
}
```

I receive the following error from salvo: `get response data failed error=Other("response does not have an upgrade extension")`
Salvo doesn't allow setting the ws as protocol. Which results in `"ws", cannot_be_a_base`

Any pointers would be helpful

endpoint im trient to proxy: ws://xxxx/:/websockets/notifications

**Comment by chrislearn** [comment]
Fixed in version 0.49.1
